network/network.go

```diff
@@ -92,9 +94,12 @@ func (n *Network) Configure(config core.ServerConfig) error {
 	n.graph = dag.NewBBoltDAG(db, dag.NewSigningTimeVerifier(), dag.NewPrevTransactionsVerifier(), dag.NewTransactionSignatureVerifier(n.keyResolver))
 	n.payloadStore = dag.NewBBoltPayloadStore(db)
 	n.publisher = dag.NewReplayingDAGPublisher(n.payloadStore, n.graph)
-	peerID := p2p.PeerID(uuid.New().String())
-	n.protocol.Configure(n.p2pNetwork, n.graph, n.publisher, n.payloadStore, time.Duration(n.config.AdvertHashesInterval)*time.Millisecond, peerID)
-	networkConfig, p2pErr := n.buildP2PConfig(peerID)
+	n.peerID = p2p.PeerID(uuid.New().String())
+	n.protocol.Configure(n.p2pNetwork, n.graph, n.publisher, n.payloadStore, n.collectDiagnostics,
```

Comment:
Yeah I noticed... Could indicate the type is doing too much. I think fixing that would mean quite a big refactor of how the protocol is implemented (it's really no-nonsense, straight coding right now). Might have to look how other systems implement their protocol handling to avoid a God-class like we're creating now. Without DI of course.

Reply:
Discussed with @woutslakhorst, made an issue for it so it can be refactored later: https://github.com/nuts-foundation/nuts-node/issues/330